Action does not support being run multiple times from the same workflow

## Describe the bug

The root cause of this is a variation of #631, but I wanted to file a separate issue since the symptom/use case is different.

Today, the action unilaterally adds a check run with a specific name ("Accessibility Checks") to report a summary of results, but if you have two different jobs in the same workflow both attempting to report results simultaneously, they can get confused and give you a confusing error when they both try to write to the same check, like in [this example](https://github.com/microsoft/accessibility-insights-action/runs/5726864137?check_suite_focus=true).

<details>
<summary>log output from a task failing in this way</summary>

```
Creating check run with status as in progress
Started local web server. Url: http://localhost:50381 Root directory: D:\a\accessibility-insights-action\accessibility-insights-action/dev/website-root
Scanning URL c:/Program%20Files/Git/
  Error: [Exception]ErrorWithCause: An error occurred while scanning website page c:/Program%20Files/Git/
      at Logger.trackExceptionAny (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:73815:29)
      at Scanner.<anonymous> (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:74874:29)
      at Generator.next (<anonymous>)
      at fulfilled (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:74799:58)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
  caused by: TypeError: Cannot read property 'baseUrl' of undefined
      at Scanner.getCombinedReportParameters (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:74887:60)
      at Scanner.<anonymous> (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:748[66](https://github.com/microsoft/accessibility-insights-action/runs/5726864137?check_suite_focus=true#step:6:66):55)
      at Generator.next (<anonymous>)
      at fulfilled (D:\a\accessibility-insights-action\accessibility-insights-action\packages\gh-action\dist\index.js:[74](https://github.com/microsoft/accessibility-insights-action/runs/5726864137?check_suite_focus=true#step:6:74)[79](https://github.com/microsoft/accessibility-insights-action/runs/5726864137?check_suite_focus=true#step:6:79)9:58)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
  Updating check run with status as failed
  Accessibility scanning of URL c:/Program%20Files/Git/ completed
  Error: Process completed with exit code 1.
```

</details>



**To Reproduce**
Steps to reproduce the behavior:

1. Create a workflow which runs multiple jobs that each invoke the action. For example, a job with a strategy matrix:

```yaml
jobs:
  example-job:
    strategy:
      matrix:
        os: ["ubuntu-latest", "windows-latest"]
    runs-on: ${{ matrix.os }}
    steps:
    # ...
    - name: Scan test site
      uses: accessibility-insights-action@v2
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        site-dir: ${{ github.workspace }}/dev/website-root
```

2. Run it
3. Observe the failure above

## Expected behavior

Running a scan from within a matrix strategy should "just work" with no special extra work from users.

## Screenshots

n/a

## Context (please complete the following information)

- OS Name & Version: n/a
- AI-Action Version & Environment: Dev build from main as of writing, @[b6d3472](https://github.com/microsoft/accessibility-insights-action/commit/b6d34720756607fbfac9ab0cc3e3cd47ae375165)
- Browser Version: n/a
- Target Page: n/a

## Are you willing to submit a PR?

sure

## Did you search for similar existing issues?

yes

## Additional context

This will probably be solved incidentally by the same work that resolves #631

--- a/.github/workflows/self-test.yml
+++ b/.github/workflows/self-test.yml
@@ -15,7 +15,6 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       checks: write
-      pull-requests: write
     steps:
     - name: Checkout repo
       uses: actions/checkout@v2

--- a/.github/workflows/self-test.yml
+++ b/.github/workflows/self-test.yml
@@ -12,10 +12,7 @@ on:
 jobs:
   self-test-dev:
     name: Self-test GitHub Action
-    strategy:
-      matrix:
-        os: ["ubuntu-latest", "windows-latest"]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     permissions:
       checks: write
       pull-requests: write

